api/v1alpha1/nodepool_types.go

```diff
@@ -89,12 +89,17 @@ type NodePoolSpec struct {
 	// +immutable
 	Platform NodePoolPlatform `json:"platform"`
 
-	// NodeCount is the desired number of nodes the pool should maintain. If
-	// unset, the default value is 0.
+	// Deprecated: Use Replicas instead
```

Comment:
nit: "Deprecated: Use Replicas instead. NodeCount will be dropped in the next API release."